### PR TITLE
Bump nonce annotation for local path function create and update

### LINKED ISSUE #1127: Function update command should trigger new revision even for local path
If I run `riff function update upper --local-path .` on local source directory the configuration isn't modified so no new revision is created. The image is pushed but we need to force a new revision to be created.

### PR DESCRIPTION
**Pull request by trisberg**
- rename `buildAnnotation` to `nonceAnnotation`

Fixes #1127

### CODE REVIEW

File: pkg/core/function.go [hunk 1]
```diff
@@ -550,6 +551,7 @@ func (c *client) UpdateFunction(buildpackBuilder Builder, options UpdateFunction
 		if err != nil {
 			return err
 		}
+		c.bumpNonceAnnotationForRevision(service)
```
Comment (ericbottard) [inline pkg/core/function.go]: Isn't that already taken care of by line 525 (previously 524)?

File: pkg/core/function.go [hunk 2]
```diff
@@ -110,6 +110,7 @@ func (c *client) CreateFunction(buildpackBuilder Builder, options CreateFunction
 				return nil, nil, err
 			}
 		}
+		c.bumpNonceAnnotationForRevision(s)
```
[the remark this answers is not in the document]
Comment (trisberg) [reply pkg/core/function.go]: there are different annotations for builds and revisions, separate objects, we should probably move the `c.bumpNonceAnnotationForBuild(s)` call to the cluster build section

File: pkg/core/function.go (hunk 1 above)
Comment (trisberg) [reply pkg/core/function.go]: same as previous comment

File: pkg/core/function.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ericbottard) [reply pkg/core/function.go]: Huh, sorry I totally misread that.

**Comment by scothis** [comment]
lgtm. Should we add function update to FATS to detect these breakages in the future?

**Comment by trisberg** [comment]
I think it would be nice to add update test to FATS

File: pkg/core/annotations.go [hunk 3]
```diff
@@ -29,12 +29,12 @@ func (c *client) bumpNonceAnnotationForRevision(s *v1alpha1.Service) {
 	if annotations == nil {
 		annotations = map[string]string{}
 	}
-	build := annotations[buildAnnotation]
+	build := annotations[nonceAnnotation]
```
Comment (fbiville) [inline pkg/core/annotations.go]: Do we want to use more names including `nonce`? I believe @glyn pointed out that this word has a bad connotation in the UK.

Comment (glyn) [reply pkg/core/annotations.go]: It's not a big problem. The term has been popular in computer security circles, regardless of the dodgy connotations. Just know that any UK readers are likely to wince, or at least smile wryly, as they read this code.

Comment (trisberg) [reply pkg/core/annotations.go]: I'm fine with changing this to something other than `nonce`, I think we had `build` originally but changed it, @scothis might have some opinion on this

Comment (scothis) [reply pkg/core/annotations.go]: no strong opinion.

**Comment by ericbottard** [comment]
Merged as 008a528e

**Pull request closed without merging**